MPDX-7539 TntConnect Import Page

caleballdrin opened the pull request:
## Description
https://jira.cru.org/secure/RapidBoard.jspa?rapidView=3&view=detail&selectedIssue=MPDX-7539#

- Creating UI
- Using Rest API to upload `file` with `tag_list` and `override`
- Success modal
- Loading indicators

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

dr-bizz commented on src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:
```diff
@@ -146,11 +146,9 @@ export const ContactTags: React.FC<ContactTagsProps> = ({
                 multiple
                 freeSolo
-                autoSelect
                 autoHighlight
                 fullWidth
                 loading={loading}
-                popupIcon={<ContactTagIcon />}
                 filterSelectedOptions
                 value={tagList}
```
Can you explain this change to me? Seems like we're removing functionailty from the whole site for one page

caleballdrin replied:
We have been removing `autoSelect` throughout the codebase as we come across it because it is not user-friendly (it selects things that the user did not click on).
`popupIcon` is not currently doing anything on this component. The icon does not show anywhere.

dr-bizz commented on src/components/Tool/TntConnect/TntConnect.tsx:
```diff
@@ -0,0 +1,392 @@
+  const { appName } = useGetAppSettings();
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: sessionData } = useSession();
```
Since you're using the backend, you don't need to call useSession on the frontend to pass the apiToken to the NextJs backend as ou can get it there.

dr-bizz commented on src/components/Tool/TntConnect/TntConnect.tsx:
```diff
@@ -0,0 +1,392 @@
+    }
+    resetForm();
+    setTntFile(null);
+    enqueueSnackbar(t('Upload Complete'), {
```
Do you want to mention something about how their upload is being processed and give it time for MPDX to update?

caleballdrin replied:
The modal pops up and explains that it will take time for their upload to process.

dr-bizz commented on src/components/Tool/TntConnect/TntConnect.tsx:
```diff
@@ -0,0 +1,392 @@
+                  <Box sx={{ display: 'flex', alignItems: 'center' }}>
+                    <Button
+                      component="label"
+                      role={undefined}
```
Do you want the role to be undefined?

dr-bizz reviewed (changes requested):
Great work on this!

Let's move all the NextJs backend upload files into one folder called `/uploads`
And let's do the same on the front-end with the uploadTntConnect and upload avatar TS files.

dr-bizz commented on pages/api/upload-tnt-connect-import.page.ts:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    const {
+      fields: { override, selectedTags, accountListId, apiToken },
```
```
const jwt = await getToken({
      req,
      secret: process.env.JWT_SECRET,
    });
    const apiToken = jwt?.apiToken;
    if (!apiToken) {
      res.status(401).send('Unauthorized');
      return;
    }
    ```

dr-bizz replied:
`pages/api/upload-person-avatar.page.ts`

caleballdrin replied:
ahh right. thanks.

dr-bizz reviewed (changes requested):
Great work on this! Everything looks great, I did spot 2 issues. I don't know how much you can control though, as it looks like they could be API issues.

The first issue: I added a tag to all contacts when they imported. When I went to the contact, the tag read as "undefined". **Screenshot 1**

The second issue: I imported the TntConnect file, and removed the country of one of the contacts. I then re-imported the file but asked it to "only fill blank fields in current contacts and/or add new contacts." However, the contact's country was never filled in.

_On doing the second import, I added a different tag to the contacts. This never got applied._

![Screenshot 2024-06-19 at 4 39 22 PM](https://github.com/CruGlobal/mpdx-react/assets/56281168/401ea5da-12e6-43da-b539-3a5b2c2efd32)

![Screenshot 2024-06-19 at 4 42 20 PM](https://github.com/CruGlobal/mpdx-react/assets/56281168/c6202174-661f-4b2b-90a5-f0fc5d98eacf)

caleballdrin commented:
> Great work on this! Everything looks great, I did spot 2 issues. I don't know how much you can control though, as it looks like they could be API issues.
> 
> The first issue: I added a tag to all contacts when they imported. When I went to the contact, the tag read as "undefined". **Screenshot 1**
> 
> The second issue: I imported the TntConnect file, and removed the country of one of the contacts. I then re-imported the file but asked it to "only fill blank fields in current contacts and/or add new contacts." However, the contact's country was never filled in.
> 
> _On doing the second import, I added a different tag to the contacts. This never got applied._
> 
> ![Screenshot 2024-06-19 at 4 39 22 PM](https://private-user-images.githubusercontent.com/56281168/341200229-401ea5da-12e6-43da-b539-3a5b2c2efd32.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MzAzNTksIm5iZiI6MTcxODgzMDA1OSwicGF0aCI6Ii81NjI4MTE2OC8zNDEyMDAyMjktNDAxZWE1ZGEtMTJlNi00M2RhLWI1MzktM2E1YjJjMmVmZDMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDIwNDczOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiMjkxZjM1MzQ3ODBlMjBhNGYzNjdkY2I5Mjk0YTU1NTJmN2YxYzdmZGZhN2NmMjc1YzFlYmY4OTQ5MjQxMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QZqDA0FxpvkcnPeukheCyrJ6BdW606iGurbi6P8hTWM)
> 
> ![Screenshot 2024-06-19 at 4 42 20 PM](https://private-user-images.githubusercontent.com/56281168/341201175-c6202174-661f-4b2b-90a5-f0fc5d98eacf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MzAzNTksIm5iZiI6MTcxODgzMDA1OSwicGF0aCI6Ii81NjI4MTE2OC8zNDEyMDExNzUtYzYyMDIxNzQtNjYxZi00YjJiLTkwYTUtZjBmYzVkOThlYWNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDIwNDczOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0MTdhNDAwODdmYmU1ZmY2ZjQ2M2MzYTYwOTZmNzM3YWI5MWI4NWQ2MDA0OTZjOWRkOTc4NzlmZjQyODkwZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J9ZsPFAMVBDiVWLAm4FdLwNPx_R5ewJoK1JBUu4PcbM)

@dr-bizz I just fixed the error with the tags. For overwriting data, the API might not compare each of the address fields. Only the whole address. I just tried overwriting and it worked for me. I had deleted the address and the person and it added it back in. It kept my existing note and added to it. I had deleted the address and the person and it added it back in.

dr-bizz reviewed (approved):
Looks great! 🔥🔥
